tests/std/tests/P0896R4_views_drop_while/test.cpp

```diff
@@ -0,0 +1,431 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <forward_list>
+#include <ranges>
+#include <span>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+
+// Test a silly precomposed range adaptor pipeline
+template <int X>
+constexpr auto is_less_than = [](const auto& x) { return x < X; };
+
+using Pred = remove_const_t<decltype(is_less_than<3>)>;
+STATIC_ASSERT(is_nothrow_copy_constructible_v<Pred>&& is_nothrow_move_constructible_v<Pred>);
+
+constexpr auto pipeline = views::drop_while(is_less_than<3>) | views::drop_while(is_less_than<3>)
+                          | views::drop_while(is_less_than<3>) | views::drop_while(is_less_than<3>);
+
+template <class Rng, class V = views::all_t<Rng>>
+using pipeline_t = ranges::drop_while_view<
+    ranges::drop_while_view<ranges::drop_while_view<ranges::drop_while_view<V, Pred>, Pred>, Pred>, Pred>;
+
+template <class Rng>
+concept CanViewDrop_while = requires(Rng&& r) {
+    views::drop_while(static_cast<Rng&&>(r), is_less_than<3>);
+};
+
+template <ranges::input_range Rng, ranges::random_access_range Expected>
+constexpr bool test_one(Rng&& rng, Expected&& expected) {
+    using ranges::drop_while_view, ranges::bidirectional_range, ranges::common_range, ranges::contiguous_range,
+        ranges::enable_borrowed_range, ranges::forward_range, ranges::iterator_t, ranges::prev,
+        ranges::random_access_range;
+
+    constexpr bool is_view = ranges::view<remove_cvref_t<Rng>>;
+
+    using V = views::all_t<Rng>;
+    using R = drop_while_view<V, Pred>;
+    STATIC_ASSERT(ranges::view<R>);
+    STATIC_ASSERT(ranges::input_range<R>);
+    STATIC_ASSERT(forward_range<R> == forward_range<Rng>);
+    STATIC_ASSERT(bidirectional_range<R> == bidirectional_range<Rng>);
+    STATIC_ASSERT(random_access_range<R> == random_access_range<Rng>);
+    STATIC_ASSERT(contiguous_range<R> == contiguous_range<Rng>);
+
+    // Validate range adaptor object and range adaptor closure
+    constexpr auto drop_while_even = views::drop_while(is_less_than<3>);

```

Comment:
The name `drop_while_even` is incorrect (a copy-paste relic from `filter_even`).

Reply:
I'll change to `closure`.